source/docs/advanced/theme/wysiwyg.md

```diff
@@ -4,116 +4,385 @@ title: Display WYSIWYG content
-
-    if (!isExternal) {
-      return (
-        <Link to={target} external={isExternal}>
-          {node.children.map(convertNodeToElement)}
-        </Link>
-      );
-    }
+1. (Server side) Add the product data to your GraphQL schema
+    1. Make sure that your dependencies are up to date in your GraphQL module
+```diff
+// my-module/server/modules/wysiwyg/index.js
+export default {
+  namespace: "Magento1/Cms",
++  dependencies: [
++    "Front-Commerce/Wysiwyg", // to make sure that Wysiwyg related features are available
++    "Magento2/Catalog/Product" // to make sure that you can fetch a product in your Wysiwyg data
++  ],
+```
+    2. Define a new `WysiwygProductNameData` type that implements `WysiwygNodeData` and expose all the data you need to display your component
+```graphql
+# my-module/server/modules/wysiwyg/schema.gql
+type WysiwygProductNameData implements WysiwygNodeData {
+  dataId: ID
+  product: Product
+}
+```
+    3. Setup the resolvers to tell GraphQL how to fetch the `product` field in your new `WysiwygProductNameData` type
+```diff
+// my-module/server/modules/wysiwyg/resolvers.js
+export default {
++  WysiwygProductNameData: {
++    product: ({ node }, _, { loaders }) => {
++      const productSkuAttribute = node.attrs.find(({ name }) => name === "sku")
++      if (!productSkuAttribute) {
++        return null;
++      }
++      return loaders.Product.load(productSkuAttribute.value)
++    }
++  },
+}
+```
+    4. Register the HTML tag and associate it with the GraphQL type
+```diff
+// my-module/server/modules/wysiwyg/index.js
+contextEnhancer: ({ loaders }) => {
++  loaders.Wysiwyg.registerNodeType(
++    // the tag name in your HTML
++    "product-name",
++    // The associated GraphQL type name
++    "WysiwygProductNameData"
++  );
+}
+```
+2. (Client side) Fetch the data to make it available in your custom Product Name component
+    1. Override the `theme/modules/WysiwygV2/WysiwygFragment.gql` to fetch the new `WysiwygProductNameData`
+```diff
+#import "./MagentoWysiwyg/MagentoWysiwygFragment.gql"
+
+fragment WysiwygFragment on Wysiwyg {
+  childNodes
+  data {
+    dataId
++   ... on WysiwygProductNameData {
++     product {
++       sku
++       name
++     }
++   }
   }
-};
+  ... on MagentoWysiwyg {
+    ...MagentoWysiwygFragment
+  }
+}
 ```
+    Feel free to split this in a Fragment dedicated to your `CustomWysiwyg` just like it is done in the core for `MagentoWysiwyg`.
+    2. Reference the `product-name` tag in your componentMap and use the new `data` prop available in the mapped component.
+```diff
+const defaultComponentsMap = {
++  product-name: (props) => <span>{props.data.product.name}</span>
+}
+```
+    <blockquote className="important">
+    <b>Note:</b> Please keep in mind that for simplicity sake, the product name component is not code split, but if it grows in complexity, please make sure to code split it using <code>@loadable/component</code> to avoid too large bundles.
```

Comment:
```suggestion
    <b>Note:</b> Please keep in mind that for simplicity's sake, the product name component is not code splitted, but if it grows in complexity, please make sure to code split it using <code>@loadable/component</code> to avoid too large bundles.
```

Reply:
Not sure about splitted. https://www.usingenglish.com/reference/irregular-verbs/split.html